cpp/include/cudf/contiguous_split.hpp

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/table/table.hpp>
+#include <cudf/types.hpp>
+
+#include <memory>
+#include <vector>
+
+namespace cudf {
+
+/**
+ * @addtogroup column_copy
+ * @{
+ * @file
+ * @brief Table APIs for contiguous_split, pack, unpack, and metadadata
+ */
+
+/**
+ * @brief Column data in a serialized format
+ *
+ * @ingroup copy_split
```

Comment:
I don't think we typically use such tag in cudf code.
```suggestion
```

Reply:
Since I am just moving code, I carried the doxygen groups because the old code used them (https://github.com/rapidsai/cudf/blob/branch-23.06/cpp/include/cudf/copying.hpp#L629)

When running doxygen you do get a module under Copying/Slicing that would group the logical slice `cudf::slice` and `cudf::contiguous_split` together as split operations (even though they are in separate headers), which is nice. I also see all kinds of other doxygen modules for factories, aggregates, scalar operations, lists, utilities, etc. It seems to be used either with `@ingroup` or `@addtogroup`.

I'd like to keep that indexing but if you think cuDF is moving away from this or the doxygen docs then I am happy to clean things up. We may want to add issues to clean up other parts of cuDF as well in that case.

Reply:
The current usage of `@ingroup` and `@addtogroup` are to organize the Modules page as well as identify which declarations go with what header file. Please keep the indexing and tag usage as you described and implemented here.